CMakeLists.txt

```diff
@@ -25,7 +25,13 @@ endif ()
 #set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native")
 option(EnableAvx "Enable AVX codepaths instead of SSE4" OFF)
 
-if (EnableAvx)
+# Automatically detect if building on an ARM based system such as the Apple M1.
+# It will replace SSE functions with ARM NEON functions using sse2neon.h from 
+# https://github.com/DLTcollab/sse2neon, redistributable under the MIT License.
+
+if(${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm" AND ${CMAKE_SYSTEM_NAME} MATCHES "Darwin") 
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_ARM_ -march=armv8-a")
```

Comment:
sse2neon [suggests](https://github.com/DLTcollab/sse2neon#usage) using `-march=armv8-a+fp+simd+crypto+crc`, should we not follow this suggestion?

I guess cross-compiling for ARM M1 will be an edge case that is not worth sorting out yet? If that's so, we might be able to use `-march=native` :thinking: